pg/array.go

```diff
@@ -0,0 +1,118 @@
+package pg
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/elgris/sqrl"
+)
+
+// Array converts value into Postgres Array
+func Array(arr interface{}) sqrl.Sqlizer {
+	return array{arr}
+}
+
+// ToSql builds the query into a SQL string and bound args.
+func (a array) ToSql() (string, []interface{}, error) {
+	if err := checkArrayType(a.value); err != nil {
+		return "", nil, err
+	}
+
+	buf := &bytes.Buffer{}
+	marshalArray(a.value, buf)
+	return "?", []interface{}{buf.String()}, nil
+}
+
+var validElems = map[reflect.Kind]struct{}{reflect.Int: {}, reflect.Float32: {}, reflect.Float64: {}, reflect.String: {}}
+
+type array struct {
+	value interface{}
+}
+
+func checkArrayType(src interface{}) error {
+	t := reflect.TypeOf(src)
+	k := t.Kind()
+	if k != reflect.Slice {
+		return fmt.Errorf("Expected value of type slice, got %s", k)
+	}
+
+	for k == reflect.Slice {
+		t = t.Elem()
+		k = t.Kind()
+	}
+
+	if _, ok := validElems[k]; !ok {
+		return fmt.Errorf("Expected element of type int, float32, float64 or string, got: %s", k)
+	}
+	return nil
+}
+
+func marshalArray(src interface{}, buf *bytes.Buffer) {
+	v := reflect.ValueOf(src)
+	l := v.Len()
+	if l == 0 {
+		buf.WriteString("{}")
+		return
+	}
+
+	switch t := src.(type) {
+	case []int:
```

Comment:
What about whole family of `int` types? (`int32`, `int64`, etc). If we're not going to support those, it should rather be described in README, so a user of the library knows that he MUST explicitly convert his `[]int64` to `[]int` (which is going to be weird in case of systems where `int` is effectively `int32`)